app/src/main/java/nu/yona/app/utils/AppUtils.java

```diff
@@ -244,40 +246,43 @@ public static String getHourDigit(String digit) {
             }
             return Pair.create(buffer.toString(), totalTime);
         } catch (Exception e) {
-            AppUtils.throwException(AppUtils.class.getSimpleName(), e, Thread.currentThread(), null);
+            AppUtils.reportException(AppUtils.class.getSimpleName(), e, Thread.currentThread(), null);
         }
         return Pair.create(time, (long) 0);
     }
 
     /**
-     * Throw exception.
+     * Report exception.
      *
      * @param className class name where exception throws
      * @param e         Error
      * @param t         Current Thread (Thread.currentThread())
      * @param listener  DataLoadListener to update UI
      */
-    public static void throwException(String className, Exception e, Thread t, DataLoadListener listener) {
+    public static void reportException(String className, Exception e, Thread t, DataLoadListener listener) {
         ErrorMessage errorMessage = new ErrorMessage();
-        try {
-            if (e != null && e.getMessage() != null) {
-                errorMessage.setMessage(e.getMessage());
-            } else {
-                errorMessage.setMessage(YonaApplication.getAppContext().getString(R.string.error_message));
-            }
-            if (listener != null) {
-                listener.onError(errorMessage);
-            }else{
-                showErrorToast(errorMessage);
-            }
-        } catch (Exception x) {
-            // Catching to avoid application crash and showing user friendly message.
-            loge(TAG, "Exception in throwException method");
-            errorMessage.setMessage(YonaApplication.getAppContext().getString(R.string.generic_exception_message));
+        if (e != null && e.getMessage() != null) {
+            errorMessage.setMessage(e.getMessage());
+        } else {
+            errorMessage.setMessage(YonaApplication.getAppContext().getString(R.string.error_message));
+        }
+        if (listener != null) {
+            listener.onError(errorMessage);
+        } else {
             showErrorToast(errorMessage);
+            logExceptionToCrashlytics(e);
```

Comment:
I see there is no error logging yet here, so please add it.

Reply:
Added logging to the reportException Method.